internal/app/machined/internal/phase/phase.go

```diff
@@ -52,33 +51,49 @@ func NewRunner(data *userdata.UserData) (*Runner, error) {
 
 // Run executes all phases known to a Runner.
 func (r *Runner) Run() error {
+	errCh := make(chan error)
+
 	for _, phase := range r.phases {
-		var (
-			result *multierror.Error
-			wg     sync.WaitGroup
-		)
-		wg.Add(len(phase.tasks))
-		log.Printf("[phase]: %s", phase.name)
-		go func(p *Phase) {
-			for _, task := range p.tasks {
-				defer wg.Done()
-				var f RuntimeFunc
-				if f = task.RuntimeFunc(r.mode); f == nil {
-					// A task is not defined for this runtime mode.
-					continue
-				}
-				if err := f(r.platform, r.data); err != nil {
-					result = multierror.Append(result, err)
-				}
-			}
-		}(phase)
-		wg.Wait()
-		if result != nil {
-			return result.ErrorOrNil()
-		}
+		r.runPhase(phase, errCh)
```

Comment:
is there much difference in 
```
go r.runPhase
```
vs
```
runPhase{
  go func() {
...
  }
}
```

if there isn't, i think I'd prefer the first form so it's clear that r.runPhase is being launched as a goroutine.

Reply:
there is no difference, I agree, this is more clear